scripts/start-ku.sh

```diff
@@ -5,16 +5,23 @@ KU_TMP_DIR=/tmp/ku
 
 # Copy essential programs and scripts to /tmp, to ensure they will be available
 # during USBMS session
+KU_REQ_FILES="scripts/run-ku.sh scripts/nickel-usbms.sh bin/button_scan bin/fbink"
 mkdir -p "$KU_TMP_DIR"
-cp "/mnt/onboard/${KU_DIR}/scripts/run-ku.sh" "$KU_TMP_DIR"
-cp "/mnt/onboard/${KU_DIR}/scripts/nickel-usbms.sh" "$KU_TMP_DIR"
-cp "/mnt/onboard/${KU_DIR}/bin/button_scan" "$KU_TMP_DIR"
-cp "/mnt/onboard/${KU_DIR}/bin/fbink" "$KU_TMP_DIR"
+for cur_file in ${KU_REQ_FILES}; do
+    cp -f "/mnt/onboard/${KU_DIR}/${cur_file}" "${KU_TMP_DIR}/${cur_file##*/}"
+done
 
 # Check if we have a ku.toml file. If not, copy it from the default file
-if [ ! -e "/mnt/onboard/${KU_DIR}/config/ku.toml" ]; then
-    cp "/mnt/onboard/${KU_DIR}/config/ku.toml.default" "/mnt/onboard/${KU_DIR}/config/ku.toml"
+if [ ! -f "/mnt/onboard/${KU_DIR}/config/ku.toml" ]; then
+    cp -f "/mnt/onboard/${KU_DIR}/config/ku.toml.default" "/mnt/onboard/${KU_DIR}/config/ku.toml"
 fi
 
 cd "$KU_TMP_DIR" || exit 255
-exec ./run-ku.sh "$KU_DIR" "$KU_TMP_DIR"
+./run-ku.sh "$KU_DIR" "$KU_TMP_DIR"
+
+# Cleanup behind us, with gloves on
+# (i.e., refrain from simply doing rm -rf "${KU_TMP_DIR}" to avoid potentially unfortunate mistakes,
+# because we're root, and the rootfs is stupidly rw by default on Kobo).
```

Comment:
:+1:

Reply:
The ghost of that magnificent Steam typo lives on! :D.

Jokes aside, that's one Kindle design decision I definitely liked ;). (Well, it made/makes jailbreaking that much more annoying, but it does make a lot of sense besides that ;)).

Reply:
Especially when you realize that they decided to mount external sdcards RO, for... reasons? :D.

Reply:
I've made that mistake in an update script on my server (rm -rf "$(typoed_command)/bin"), but luckily I had it sandboxed in a mount namespace and I had backups running quite often. Ever since, I've been paranoid about removing folders, especially when complex logic/substitution is involved.

Personally, I like doing a minimal kernel + an embedded initramfs for my embedded stuff (it's really nice for network booting, swapping sd cards live, and updating with a/b boot), but I guess it's nice that Kobo doesn't do it or it would be quite a bit more tedious to modify.